jiant/scripts/download_data/datasets/xtreme.py

```diff
@@ -0,0 +1,372 @@
+            src=os.path.join(
+                bucc2018_temp_path, "bucc2018", f"{lang}-en", f"{lang}-en.training.{lang}"
+            ),
+            dst=test_other_path,
+        )
+        py_io.write_json(
+            data={
+                "task": "tatoeba",
+                "paths": {
+                    "val": {
+                        "eng": val_eng_path,
+                        "other": val_other_path,
+                        "labels": val_labels_path,
+                    },
+                    "test": {"eng": test_eng_path, "other": test_other_path},
+                },
+                "name": task_name,
+            },
+            path=os.path.join(task_config_base_path, f"{task_name}_config.json"),
+        )
+
+
+def download_tatoeba_data_and_write_config(task_data_base_path: str, task_config_base_path: str):
+    tatoeba_temp_path = py_io.create_dir(task_data_base_path, "tatoeba_temp")
+    download_utils.download_and_unzip(
+        "https://github.com/facebookresearch/LASER/archive/master.zip", tatoeba_temp_path,
+    )
+    languages_dict = {
+        "afr": "af",
+        "ara": "ar",
+        "bul": "bg",
+        "ben": "bn",
+        "deu": "de",
+        "ell": "el",
+        "spa": "es",
+        "est": "et",
+        "eus": "eu",
+        "pes": "fa",
+        "fin": "fi",
+        "fra": "fr",
+        "heb": "he",
+        "hin": "hi",
+        "hun": "hu",
+        "ind": "id",
+        "ita": "it",
+        "jpn": "ja",
+        "jav": "jv",
+        "kat": "ka",
+        "kaz": "kk",
+        "kor": "ko",
+        "mal": "ml",
+        "mar": "mr",
+        "nld": "nl",
+        "por": "pt",
+        "rus": "ru",
+        "swh": "sw",
+        "tam": "ta",
+        "tel": "te",
+        "tha": "th",
+        "tgl": "tl",
+        "tur": "tr",
+        "urd": "ur",
+        "vie": "vi",
+        "cmn": "zh",
+        "eng": "en",
+    }
+    raw_base_path = os.path.join(tatoeba_temp_path, "LASER-master", "data", "tatoeba", "v1")
+    for full_lang, lang in languages_dict.items():
+        task_name = f"tatoeba_{lang}"
+        if lang == "en":
+            continue
+        task_data_path = py_io.create_dir(task_data_base_path, task_name)
+        eng_src = os.path.join(raw_base_path, f"tatoeba.{full_lang}-eng.eng")
+        other_src = os.path.join(raw_base_path, f"tatoeba.{full_lang}-eng.{full_lang}")
+        eng_out = os.path.join(task_data_path, f"{lang}-en.en")
+        other_out = os.path.join(task_data_path, f"{lang}-en.{lang}")
+        labels_out = os.path.join(task_data_path, f"{lang}-en.labels")
+        tgts = [line.strip() for line in py_io.read_file_lines(eng_src)]
+        os.rename(src=other_src, dst=other_out)
+        idx = range(len(tgts))
+        data = zip(tgts, idx)
+        with open(eng_out, "w") as ftgt, open(labels_out, "w") as flabels:
+            for t, i in sorted(data, key=lambda x: x[0]):
+                ftgt.write(f"{t}\n")
+                flabels.write(f"{i}\n")
```

Comment:
I'm not sure what is going on here. Is this step obvious to someone familiar with the task?

Reply:
Not really, I should add an explanation for this. See also, the original preprocessing code: https://github.com/google-research/xtreme/blob/975d5ffb86be9aab0f010054b56f4204d790aa18/utils_preprocess.py#L390-L400

Tatoeba is a retrieval dataset where you have a set of sentences in English and another set in another language, and you need to match them. Also, it's a dataset that doesn't have training data, so it's pretty much evaluation only. However, the dataset is distributed with the sentences in order, i.e. the retrieval pairing is the sentence order. The XTREME authors basically intentionally scramble the order by sorting one of the two sets alphabetically. I'm following their recipe, but also retaining the labels for internal scoring.

In theory, we can do away with the whole scrambling thing if we trust the users to not (accidentally) cheat, but I think this is a good compromise between having something that looks like the XTREME processed outputs while still owning the logic for preparing the data ourselves.

Reply:
Thanks for the explanation. Do you mind adding a comment based on your explanation?

Reply:
Is it possible to use the xtreme/utils_preprocess.py directly?